lib/github/ldap/domain.rb

```diff
@@ -118,7 +118,14 @@ def user?(login, search_options = {})
         options = search_options.merge \
           filter: login_filter(@uid, login),
           size: 1
-        search(options).first
+
+        if @ldap.active_directory_capability?
+          # when doing a global search for a user's DN, set the search base to blank
+          options[:base] = ""
+          @ldap.global_catalog_search(options).first
+        else
+          search(options).first
+        end
```

Comment:
Need to think this through even further. We perform the `user?` query on the `Domain` object, but a Global Catalog search is not domain-specific by definition, so that type of search doesn't belong on this object, even though it's convenient to make that decision here.